plugins/modules/pacemaker_stonith.py

```diff
@@ -0,0 +1,228 @@
+    description:
+      - Timeout period for polling the STONITH creation.
+    type: int
+    default: 300
+'''
+
+EXAMPLES = '''
+---
+- name: Create pacemaker STONITH
+  hosts: localhost
+  gather_facts: false
+  tasks:
+  - name: Create virtual-ip STONITH
+    community.general.pacemaker_stonith:
+      state: present
+      name: virtual-stonith
+      stonith_type: fence_virt
+      stonith_options:
+        - "pcmk_host_list=f1"
+      stonith_operations:
+        - operation_action: monitor
+          operation_options:
+            - "interval=30s"
+'''
+
+RETURN = '''
+cluster_stonith:
+    description: The cluster STONITH output message.
+    type: str
+    sample: ""
+    returned: always
+'''
+
+from ansible_collections.community.general.plugins.module_utils.module_helper import StateModuleHelper
+from ansible_collections.community.general.plugins.module_utils.pacemaker import pacemaker_runner
+
+
+class PacemakerStonith(StateModuleHelper):
+    module = dict(
+        argument_spec=dict(
+            state=dict(type='str', default='present', choices=[
+                'present', 'absent', 'enabled', 'disabled']),
+            name=dict(type='str', required=True),
+            stonith_type=dict(type='str'),
+            stonith_options=dict(type='list', elements='str', default=list()),
+            stonith_operations=dict(type='list', elements='dict', default=list(), options=dict(
+                operation_action=dict(type='str'),
+                operation_options=dict(type='list', elements='str'),
+            )),
+            stonith_metas=dict(type='list', elements='str'),
+            stonith_argument=dict(type='dict', options=dict(
+                argument_action=dict(type='str', choices=['before', 'after', 'group']),
+                argument_options=dict(type='list', elements='str'),
+            )),
+            agent_validation=dict(type='bool', default=False),
+            wait=dict(type='int', default=300),
+        ),
+        required_if=[('state', 'present', ['stonith_type', 'stonith_options'])],
+        supports_check_mode=True
+    )
+
+    use_old_vardict = False
+    default_state = "present"
+
+    def __init_module__(self):
+        self.runner = pacemaker_runner(self.module, cli_action='stonith')
+        self.vars.set('previous_value', self._get())
+        self.vars.set('value', self.vars.previous_value, change=True, diff=True)
+
+    def _process_command_output(self, fail_on_err, ignore_err_msg=""):
+        def process(rc, out, err):
+            if fail_on_err and rc != 0 and err and ignore_err_msg not in err:
+                self.do_raise('pcs failed with error (rc={0}): {1}'.format(rc, err))
+            out = out.rstrip()
+            return None if out == "" else out
+        return process
+
+    def _get(self):
+        with self.runner('state name', output_process=self._process_command_output(False)) as ctx:
+            return ctx.run(state='status')
+
+    def _fmt_stonith_resource(self):
```

Comment:
This file is not meant to be imported by any other. Prefixing the method names with `_` is kinda redundant in this context. In the `module_utils` file that would be valid if not necessary.